ic-agent/src/agent/agent_error.rs

```diff
@@ -127,9 +127,9 @@ pub enum AgentError {
     #[error("The status response did not contain a root key.  Status: {0}")]
     NoRootKeyInStatus(Status),
 
-    /// Could not read the replica root key.
-    #[error("Could not read the root key")]
-    CouldNotReadRootKey(),
+    /// An operation failed due to a poisoned type.
+    #[error("Could not perform the requested operation due to a poisoned type")]
+    Poisoned(),
```

Comment:
A poison state shouldn't be surfaced as an error at all. A call in the stack that does not have the ability to reset the mutex should receive it as a panic; it exists to propagate panics from one thread to another and if we do not provide the ability to recover from the other panic, the result should be unwrapped. Either way there should be no such error variant.

Reply:
Sure, this is even better. I just decided to keep the existing semantics but I do agree if we unwrap the lock it is better.